src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpCacheProxyCachedRequest.java

```diff
@@ -0,0 +1,390 @@
+    {
+        this.factory = factory;
+        this.requestHash = requestHash;
+        this.acceptReply = acceptReply;
+        this.acceptRouteId = acceptRouteId;
+        this.acceptReplyId = acceptReplyId;
+        this.supplyReceiver = factory.router::supplyReceiver;
+        this.signaler = supplyReceiver.apply(acceptInitialId);
+    }
+
+   void onRequestMessage(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+   {
+        switch (msgTypeId)
+        {
+            case BeginFW.TYPE_ID:
+                final BeginFW begin = factory.beginRO.wrap(buffer, index, index + length);
+                onBegin(begin);
+                break;
+            case DataFW.TYPE_ID:
+                final DataFW data = factory.dataRO.wrap(buffer, index, index + length);
+                onData(data);
+                break;
+            case EndFW.TYPE_ID:
+                final EndFW end = factory.endRO.wrap(buffer, index, index + length);
+                onEnd(end);
+                break;
+            case AbortFW.TYPE_ID:
+                final AbortFW abort = factory.abortRO.wrap(buffer, index, index + length);
+                onAbort(abort);
+                break;
+        }
+   }
+
+    void onResponseMessage(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        switch(msgTypeId)
+        {
+            case WindowFW.TYPE_ID:
+                final WindowFW window = factory.windowRO.wrap(buffer, index, index + length);
+                onWindow(window);
+                break;
+            case SignalFW.TYPE_ID:
+                final SignalFW signal = factory.signalRO.wrap(buffer, index, index + length);
+                onSignal(signal);
+                break;
+            case ResetFW.TYPE_ID:
+                final ResetFW reset = factory.resetRO.wrap(buffer, index, index + length);
+                onReset(reset);
+                break;
+        }
+    }
+
+    private void onBegin(
+        BeginFW begin)
+    {
+        final OctetsFW extension = begin.extension();
+        final HttpBeginExFW httpBeginFW = extension.get(factory.httpBeginExRO::wrap);
+        final ListFW<HttpHeaderFW> requestHeaders = httpBeginFW.headers();
+
+        // count all requests
+        factory.counters.requests.getAsLong();
+        factory.counters.requestsCacheable.getAsLong();
+
+        if (DEBUG)
+        {
+            System.out.printf("[%016x] ACCEPT %016x %s [received request]\n",
+                    currentTimeMillis(), acceptReplyId, getRequestURL(httpBeginFW.headers()));
+        }
+
+        DefaultCacheEntry cacheEntry = factory.defaultCache.get(requestHash);
+        final String requestAuthorizationHeader = getHeader(requestHeaders, AUTHORIZATION);
+        cacheEntry.recentAuthorizationHeader(requestAuthorizationHeader);
+
+        etagMatched = CacheUtils.isMatchByEtag(requestHeaders, cacheEntry.etag());
+        if (etagMatched)
```

Comment:
`etagMatched` can be determined in the stream factory before creating this `HttpCacheProxyCachedRequest`

Reply:
Yep, but I though since only this class need this and we need to create cacheEntry as well I created it in here. Do you think I still need to move it to factory.